src/runtime_src/core/common/shim/buffer_handle.h

```diff
@@ -84,9 +85,29 @@ class buffer_handle
   // For now, this is only used when set_arg() is called upon an exec buf
   // BO where pos is the arg index.
   virtual void
-  bind_at(size_t /*pos*/, const buffer_handle* /*bh*/, size_t /*offset*/, size_t /*size*/)
+  bind_at(size_t pos, const buffer_handle* bh, size_t offset, size_t size)
   {
   }
+
+  // This is called to SHIM to build cmd-chain and submit to firmware.
+  virtual void
+  execute()
+  {
+  }
+
+  // This to get the state of the last cmd in a cmd-chain.
+  virtual ert_cmd_state
+  get_last_cmd_state()
+  {
+    return ERT_CMD_STATE_MAX;
+  };
+
+  // Get the state of a cmd in cmd-chain.
+  virtual ert_cmd_state
+  get_cmd_state(size_t idx)
+  {
+    return ERT_CMD_STATE_MAX;
+  }
```

Comment:
I need to understand how these are used.   I am not excited about amending buffer_handle with these methods.